easybuild/tools/package/utilities.py

```diff
@@ -105,51 +108,53 @@ def package_with_fpm(easyblock):
 
     _log.debug("The dependencies to be added to the package are: %s",
                pprint.pformat([easyblock.toolchain.as_dict()] + easyblock.cfg.dependencies()))
-    depstring = ''
+    deplist = []
     for dep in deps:
         if dep.get('external_module', False):
             _log.debug("Skipping dep marked as external module: %s", dep['name'])
         else:
             _log.debug("The dep added looks like %s ", dep)
             dep_pkgname = package_naming_scheme.name(dep)
-            depstring += " --depends %s" % quote_str(dep_pkgname)
+            deplist.extend(["--depends", dep_pkgname])
 
     # Excluding the EasyBuild logs and test reports that might be in the installdir
     exclude_files_glob = [
         os.path.join(log_path(), "*.log"),
         os.path.join(log_path(), "*.md"),
     ]
     # stripping off leading / to match expected glob in fpm
-    exclude_files_glob = [
-        '--exclude %s' % quote_str(os.path.join(easyblock.installdir.lstrip(os.sep), x))
-        for x in exclude_files_glob
-    ]
+    exclude_files_cmd = []
+    for x in exclude_files_glob:
+        exclude_files_cmd.extend(['--exclude',  os.path.join(easyblock.installdir.lstrip(os.sep), x)])
+
     _log.debug("The list of excluded files passed to fpm: %s", exclude_files_glob)
+    # use env to find FPM https://stackoverflow.com/questions/5658622/python-subprocess-popen-environment-path
     cmdlist = [
+        '/usr/bin/env',
         PKG_TOOL_FPM,
         '--workdir', workdir,
-        '--name', quote_str(pkgname),
-        '--provides', quote_str(pkgname),
+        '--name', pkgname,
+        '--provides', pkgname,
         '-t', pkgtype,  # target
         '-s', 'dir',  # source
         '--version', pkgver,
         '--iteration', pkgrel,
-        '--description', quote_str(easyblock.cfg["description"]),
-        '--url', quote_str(easyblock.cfg["homepage"]),
+        '--description', easyblock.cfg["description"],
+        '--url', easyblock.cfg["homepage"],
     ]
-    cmdlist.extend(exclude_files_glob)
+    cmdlist.extend(exclude_files_cmd)
 
     if build_option('debug'):
         cmdlist.append('--debug')
-
+    cmdlist.extend(deplist)
     cmdlist.extend([
-        depstring,
         easyblock.installdir,
         easyblock.module_generator.get_module_filepath(),
     ])
     cmd = ' '.join(cmdlist)
+    print "cmd: %s" % cmdlist
     _log.debug("The flattened cmdlist looks like: %s", cmd)
-    run_cmd(cmd, log_all=True, simple=True)
+    run_cmd(cmdlist, log_all=True, simple=True, cache=False, shell=False)
```

Comment:
the cache doesn't work (in that it tries to hash a list, I will try to fix that, but this command is also not cacheable).